File: LeetCode/Medium/3719-longest-balanced-subarray-i/3719-longest-balanced-subarray-i.go
func longestBalanced(nums []int) int {
	n := len(nums)
	maxLen := 0
	for i := 0; i < n-1; i++ {
		odd := make(map[int]bool)
		even := make(map[int]bool)
		for j := i; j < n; j++ {
			if nums[j]%2 == 0 {
				even[nums[j]] = true
			} else {
				odd[nums[j]] = true
			}
			if len(odd) == len(even) {
				currLen := j - i + 1
				if currLen > maxLen {
					maxLen = currLen
				}
			}
		}
	}
	return maxLen
}

File: LeetCode/Medium/3719-longest-balanced-subarray-i/3719-longest-balanced-subarray-i.cpp
class Solution {
public:
    int longestBalanced(vector<int>& nums) {
        int ans = 0, n = nums.size();
        set<int> odd;
        set<int> even;
        for (int j = 0; j < n - 1; j++) {
            for (int i = j; i < n; i++) {
                if (nums[i] % 2 == 0) {
                    even.insert(nums[i]);
                } else {
                    odd.insert(nums[i]);
                }
                if (even.size() == odd.size()) {
                    // cout << j << " " << i << " " << i - j + 1 << endl;
                    ans = max(ans, i - j + 1);
                }
            }
            odd.clear();
            even.clear();
        }
        return ans;
    }
};